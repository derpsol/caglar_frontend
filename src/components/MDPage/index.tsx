import React, { useEffect, useState } from 'react';
import marked from 'marked';
import Container from '../Container';

export interface MDPageProps {
  fileName: string;
}

const MDPage: React.FC<MDPageProps> = ({ fileName }) => {
  const [contents, setContents] = useState<string>('');

  useEffect(() => {
    const loadFile = async () => {
      const file = await import(`../../data/tabs/${fileName}.md`);
      const response = await fetch(file.default);
      const text = await response.text();
      setContents(text);
    };
    loadFile();
  }, [fileName]);

  return (
    <Container title={fileName} seo={{ title: fileName }}>
      <div dangerouslySetInnerHTML={{ __html: marked(contents) }}></div>
    </Container>
  );
};

export default MDPage;
