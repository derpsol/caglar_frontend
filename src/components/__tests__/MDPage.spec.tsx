import React from 'react';
import { render } from '@testing-library/react';
import MDPage from '../MDPage';

describe('<MDPage />', () => {
  test('should render correctly', () => {
    const { asFragment } = render(<MDPage fileName="skills" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
