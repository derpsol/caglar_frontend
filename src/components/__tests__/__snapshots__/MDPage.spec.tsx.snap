// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MDPage /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="root-0-2-1"
  >
    <h2
      class="title-0-2-2"
    >
      skills
    </h2>
    <h3
      class="subtitle-0-2-3"
    />
    <div
      class="content-0-2-4"
    />
  </div>
</DocumentFragment>
`;
